### PR TITLE
Feature/auth

### PR DESCRIPTION
**Pull request by amirfakhrullah**
- login, signup and logout endpoints working
- swagger docs available in dev route: `/api/docs`
- `api/auth/check` for quick checking if the user is logged in or not

### CODE REVIEW

**Review by cherylli — COMMENTED** [review]
just checking, are we doing auth without token?
/login endpoint response is inconsitent in the docs and the code

**Comment by abdulqshabbir** [comment]
> just checking, are we doing auth without token? /login endpoint response is inconsistent in the docs and the code

Hi Cheryll, I was having the same issue. For now the workaround for the signup route is to instead make a POST request to`/auth/signup` and pass `name` into the request body instead of `username`. I'll leave some comments for @amirfakhrullah and he'll be able to update the API.

Also, you do not need to pass in a token. I believe the server will set a cookie on the client and the client will send this cookie with each request to maintain a session. After the session is over (on logout) the cookie will be invalidated and the client won't be able to access protected resources. I'm not 100% sure about this though and @amirfakhrullah could probably elaborate.

Also for the other routes, for now you can drop the `/api` and it should work fine. We'll synchronize the api and docs soon :)

File: server/src/index.ts [hunk 1]
```diff
@@ -55,10 +56,26 @@ app.use(passport.initialize());
...
+    })
+  );
+}
+
 // routes
 app.get("/", (_req: Request, res: Response) => {
-  res.send("Express and Typescript working! Welcome to Notum backend!");
+  res.send("Welcome to Notum backend!");
 });
+app.use("/auth", authRouter);
```
Comment (abdulqshabbir) [inline server/src/index.ts]: change `app.use("/auth", authRouter)` to `app.use("/api/auth", authRouter)` to be consistent with docs

File: server/src/docs.yaml [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+          application/json:
+            schema:
+              required:
+                - email
+                - name
+                - password
+              properties:
+                email:
+                  type: string
+                username:
```
Comment (abdulqshabbir) [inline server/src/docs.yaml]: change `username` on line 27 to `name` so it's consistent with API.
Thanks btw for setting up swagger docs! It looks really cool! And I think it'll really help the front end team.

**Comment by cherylli** [comment]
> 

thanks. it works without /api but was unable to create a user without reason. Maybe nice to add why the user create was unsuccessful 

I guess we will be doing something like this? https://www.freecodecamp.org/news/authenticate-users-node-app/

i've mostly used jwt tokens, so I wasn't sure. I'll have to look it up on the frontend part of this implementation

**Comment by amirfakhrullah** [comment]
@cherylli @abdulqshabbir I have changed all the API responses. Now it will have a consistent response. It should be something like this:

```typescript
{
    success: true,
    message: "User successfully login",
    data: {
        id: "6302ed8b0f9215bcabc4593e",
        name: "amir",
        email: "amir@g.com"
    }
}
```

The `success` and `message` will be required for all the API responses (except `/api/auth/check`).

We will be using cookie-session as our auth so that the frontend doesn't need to assign authorization header for every request they make to the backend.

I have created a GET endpoint `/api/auth/check` where frontend can use to check if the user is logged in or not. The response would look like this:
```typescript
{
    isLoggedIn: true,
    user: {
        id: "6302ed8b0f9215bcabc4593e",
        name: "amir",
        email: "amir@gmail.com"
    }
}
```
This endpoint should be used on the frontend to pre-check before rendering which routes need to display (`AuthRoutes` or `AppRoutes`).

**Comment by 12cassie34** [comment]
Looks good to me.

**Comment by amirfakhrullah** [comment]
## Latest changes

@abdulqshabbir 

- I fixed some folder structures, and refactor where possible
- I realised `pretty-quick` skip some files when reformat, so I use `prettier --write` instead

**Comment by amirfakhrullah** [comment]
New Commit: 07b596f552b6d77a0a531d2c8ec0005b78f469c0
- Add helper functions that will be used to filter lesson cards for the week and sort by time
- Update model schemas to align with the helper functions

**Comment by abdulqshabbir** [comment]
> ## Latest changes
> 
> @abdulqshabbir
> 
>     * I fixed some folder structures, and refactor where possible

Love the refactoring of the zod validators!

**Comment by abdulqshabbir** [comment]
> New Commit: [07b596f](https://github.com/chingu-voyages/v40-bears-team-37/commit/07b596f552b6d77a0a531d2c8ec0005b78f469c0)
> 
>     * Add helper functions that will be used to filter lesson cards for the week and sort by time
> 
>     * Update model schemas to align with the helper functions

Helper functions look good. Massaging the data looks like it was really complicated :)

**Review by abdulqshabbir — APPROVED** [review]
Looks good!

**Pull request merged**